### PR TITLE
Update PlatformApis.cs - Update based on CR comments

### PR DESCRIPTION
**Pull request by dseefeld**
Missed @eerhardt review comments.  Adding them here.

### CODE REVIEW

File: tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj [hunk 1]
```diff
@@ -2,15 +2,16 @@
   <PropertyGroup>
     <TargetFramework>netstandard1.5</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <CoreSetupPlatformAbstractionsPath>../../../src/core-setup/src/managed/Microsoft.DotNet.PlatformAbstractions/</CoreSetupPlatformAbstractionsPath>
```
Comment (weshaggard) [inline tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj]: Please use `$(ProjectDir)src/core-setup/src/managed/Microsoft.DotNet.PlatformAbstractions/` That way we don't easily break the relative paths by moving this project.

File: tools-local/init-build.proj [hunk 2]
```diff
@@ -26,7 +26,7 @@
   </Target>
 
   <Target Name="BuildTasks">
-    <Exec Command="$(DotNetCliToolDir)dotnet build tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj" />
+    <Exec Command="$(DotNetCliToolDir)dotnet build tasks\Microsoft.DotNet.SourceBuild.Tasks\Microsoft.DotNet.SourceBuild.Tasks.csproj /p:SubmoduleDirectory=$(SubmoduleDirectory)" />
```
Comment (weshaggard) [inline tools-local/init-build.proj]: Why did you need to pass it in explicitly?

Comment (dseefeld) [reply tools-local/init-build.proj]: I was getting a weird failure after I updated it to include dir.props and so I thought maybe I was getting more than I wanted by including all of dir.props, so I switched it to just include the directory directly.  I'm going to switch it back and see if I can figure out the errors.

File: tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj [hunk 3]
```diff
@@ -1,16 +1,22 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <!-- We use dotnet-cli to build BuildTasks, so we need to disable BuildTools Roslyn properties. -->
+    <DisableBuildToolsRoslynVersion>true</DisableBuildToolsRoslynVersion>
```
Comment (weshaggard) [inline tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj]: It would seem as though we may need to split dir.props up a little bit if we cannot easily import it in our projects.

Comment (eerhardt) [reply tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj]: We do the same thing in `core-setup`.

With the move to using the .NET SDK's MSBuild work that Jose is doing, this should get better with time.  Eventually this can just go away.

File: dir.props [hunk 4]
```diff
@@ -36,8 +36,7 @@
   <PropertyGroup>
     <BaseOutputPath>$(ProjectDir)bin/</BaseOutputPath>
     <ToolsDir>$(ProjectDir)Tools/</ToolsDir>
-    <TaskDirectory>$(ProjectDir)tools-local/tasks/</TaskDirectory>
-    <TasksBinDir>$(TaskDirectory)Microsoft.DotNet.SourceBuild.Tasks/bin/Debug/netstandard1.5/</TasksBinDir>
+    <TasksBinDir>$(ProjectDir)bin/AnyCPU/Debug/netstandard1.5/</TasksBinDir>
```
Comment (weshaggard) [inline dir.props]: Isn't that going to be the wrong output for all other projects?

Comment (dseefeld) [reply dir.props]: No.  This should be correct for all projects.

Comment (weshaggard) [reply dir.props]: We are now putting the task dll directly in that directory? It seems a little odd that we no longer put it under a directory with it's name in the path.

Comment (dseefeld) [reply dir.props]: I'm going to update OutputPath to put it back under the tasks directory.

File: tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj [hunk 5]
```diff
@@ -1,16 +1,23 @@
...
+  </PropertyGroup>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))/dir.props" />
   <PropertyGroup>
     <TargetFramework>netstandard1.5</TargetFramework>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <CoreSetupPlatformAbstractionsPath>$(SubmoduleDirectory)core-setup/src/managed/Microsoft.DotNet.PlatformAbstractions/</CoreSetupPlatformAbstractionsPath>
+    <OutputPath>$(MSBuildThisFileDirectory)bin/$(Configuration)</OutputPath>
```
Comment (eerhardt) [inline tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj]: I thought we wanted all build output to go into the root `bin` directory, and not into `bin` directories at each directory level?

Comment (weshaggard) [reply tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj]: good point, we should scope it to under the root bin folder.

Comment (dseefeld) [reply tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/Microsoft.DotNet.SourceBuild.Tasks.csproj]: This is where it was located before I made the change.  Including dir.props caused it to move, which caused other issues with usage of tasks and building the tarball.  I'm putting it back where it was before.
